src/Symfony/Component/Ldap/Connection/Connection.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Ldap\Connection;
+
+use Symfony\Component\Ldap\Exception\ConnectionException;
+use Symfony\Component\Ldap\Exception\LdapException;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Charles Sarrazin <charles@sarraz.in>
+ */
+class Connection implements ConnectionInterface
+{
+    private $bound = false;
+    private $connection;
+    private $config;
```

Comment:
True. However, one of the main questions is:

> how to handle multiple LDAP connections?

I see three possibilities:

* Either we encapsulate n `Connection` objects in a registry. The registry would be an additional class, implementing `ConnectionInterface`. Whether the registry actually extends the `Connection` class is mostly a matter of code reuse.
* Or we encapsulate n ldap connection resources in a single `Connection` object. In this case, the `Connection::getConnection()` method would return an array of Ldap connection resources.
* Or we make the Ldap client aware that there may be multiple connections.

Even though the interface is still the same, the main issue with the `ldap` extension is that you need to transfer resources from one place to another (you need the `connection` resource in `ldap_search()`, which returns a `search` resource, used in `ldap_get_entries()`, and `ldap_next_entry`, etc.).

Because we actually need to expose the resources, I'm not even sure we should have a `ConnectionInterface`.

The `Connection`'s configuration could actually be moved outside, in the `Ldap` class' logic.